data/contract_docs/heatmap.md

```diff
@@ -0,0 +1,232 @@
+      <p>
+        <strong>Name: Time</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: </strong>
+      </p>
+      <p>
+        <strong>Labels: {"le":<em> "10"</em>}</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: </strong>
+      </p>
+      <p>
+        <strong>Labels: {"le":<em> "20"</em>}</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: </strong>
+      </p>
+      <p>
+        <strong>Labels: {"le":<em> "+Inf"</em>}</strong>
+      </p>
+    </td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>6</td>
+    <td>7</td>
+    <td>8</td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>6</td>
+    <td>7</td>
+    <td>8</td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>6</td>
+    <td>7</td>
+    <td>8</td>
+  </tr>
+</table>
+
+
+Note: [Timeseries wide](./timeseries.md#time-series-wide-format-timeserieswide) can be used directly
+as heatmap-buckets, in this case each value field becomes a row in the heatmap.
+
+
+## Heatmap scanlines (HeatmapScanlines)
+
+In this format, each row in the frame indicates the value of a single cell in a heatmap.
+There exists a row for every cell in the heatmap.
+
+**Example:**
+
+<table>
+  <tr>
+    <td>
+      <strong>Type: Time</strong>
+      <p>
+        <strong>Name: xMax|xMin|x</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: yMax|yMin|y</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: Count</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: Total</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: Speed</strong>
+      </p>
+    </td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>100</td>
+    <td>1</td>
+    <td>1</td>
+    <td>1</td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>200</td>
+    <td>2</td>
+    <td>2</td>
+    <td>2</td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>300</td>
+    <td>3</td>
+    <td>3</td>
+    <td>3</td>
+  </tr>
+
+  <tr>
+    <td>1653416392000</td>
+    <td>100</td>
+    <td>4</td>
+    <td>4</td>
+    <td>4</td>
+  </tr>
+  <tr>
+    <td>1653416392000</td>
+    <td>200</td>
+    <td>5</td>
+    <td>5</td>
+    <td>5</td>
+  </tr>
+  <tr>
+    <td>1653416392000</td>
+    <td>300</td>
+    <td>6</td>
+    <td>6</td>
+    <td>6</td>
+  </tr>
+</table>
+
+This format requires uniform cell sizing
+
+Note that multiple "value" fields can included to represent multiple dimensions within the same cell.  
+The first value field is used in the display, unless explicilty configured
+
+The field names for yMax|yMin|y indicate the aggregation period or the supplied values.
```

Comment:
Is this case sensitive? I don't care if it is or not, just that we pick.

Reply:
In transformations, the case sensitivity is the same, I suggest with stick with lower case y and uppercase M, yMin, yMax